crates/bevy_audio/src/audio_source.rs

```diff
@@ -43,9 +50,12 @@ impl AssetLoader for Mp3Loader {
     }
 }
 
+/// Mark a type as decodable
```

Comment:
This could use more information about what exactly it means for something to be "decodable".

Many of the readers will have 0 audio experience.

Reply:
I find it very hard to specify because this trait is so generic. There is no restriction on the Decoder type in the trait.
The restrictions come later in the implementations of AudioOutput and Audio. There we require the Decoder to yield some rodio types.

Reply:
I moved the trait bounds from the implementations to the decodable type. Now it should be a bit clearer what the trait means and we have a lot less duplication in where clauses.

This is not the same API, but I couldn't think of a use case where it would be a problem to have the rodio trait bounds on the Decodable trait directly.

What do you think?

Reply:
This will lead to a little bit of generic-pollution, but the boilerplate reduction and clarity are worth it here IMO.